### PR TITLE
Enhancement: replace @CurrentUser user type with userDTO

### PR DESCRIPTION
**Pull request by ppratikcr7**
This resolves issue #1894

### CODE REVIEW

File: backend/packages/Upgrade/src/api/repositories/ExperimentAuditLogRepository.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class ExperimentAuditLogRepository extends Repository<ExperimentAuditLog>
   public async saveRawJson(
     type: LOG_TYPE,
     data: any,
-    user: User,
+    user: UserDTO,
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/repositories/ExperimentAuditLogRepository.ts:39]: This is incorrect. You will be saving User document in the database not the DTO

File: backend/packages/Upgrade/src/api/controllers/UserController.ts [hunk 2]
```diff
@@ -172,7 +172,7 @@ export class UserController {
    */
   @Post()
   public create(
-    @Body({ validate: true }) user: UserDetailsValidator, @Req() request: AppRequest): Promise<User> {
+    @Body({ validate: true }) user: UserDTO, @Req() request: AppRequest): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/controllers/UserController.ts:175]: The return type should not be changed

File: backend/packages/Upgrade/src/api/controllers/UserController.ts [hunk 3]
```diff
@@ -215,8 +215,8 @@ export class UserController {
   @Post('/details')
   public updateUserDetails(
     @Body({ validate: true })
-    user: UserDetailsValidator
-  ): Promise<User> {
+    user: UserDTO
+  ): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/controllers/UserController.ts:219]: The return type should not be changed

File: backend/packages/Upgrade/src/api/controllers/UserController.ts [hunk 4]
```diff
@@ -241,7 +241,7 @@ export class UserController {
    *            description: Delete User By email
    */
   @Delete('/:email')
-  public delete(@Param('email') email: string): Promise<User> {
+  public delete(@Param('email') email: string): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/controllers/UserController.ts:244]: The return type should not be changed

File: backend/packages/Upgrade/src/api/services/UserService.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ export class UserService {
     this.emails = new Emails();
   }
 
-  public async upsertUser(userDTO: UserDetailsValidator, logger: UpgradeLogger): Promise<User> {
+  public async upsertUser(userDTO: UserDTO, logger: UpgradeLogger): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/UserService.ts:31]: The return type of the function is User

File: backend/packages/Upgrade/src/api/services/UserService.ts [hunk 6]
```diff
@@ -104,15 +104,15 @@ export class UserService {
     return this.userRepository.findByIds([email]);
   }
 
-  public async updateUserDetails(firstName: string, lastName: string, email: string, role: UserRole): Promise<User> {
+  public async updateUserDetails(firstName: string, lastName: string, email: string, role: UserRole): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/UserService.ts:107]: Same as above

File: backend/packages/Upgrade/src/api/services/UserService.ts [hunk 7]
```diff
@@ -104,15 +104,15 @@ export class UserService {
...
     const response = await this.userRepository.updateUserDetails(firstName, lastName, email, role);
     if (response) {
       this.sendRoleChangedEmail(email, role);
     }
     return response;
   }
 
-  public deleteUser(email: string): Promise<User> {
+  public deleteUser(email: string): Promise<UserDTO> {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/UserService.ts:115]: Same as above

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

**Comment by ppratikcr7** [comment]
@VivekFitkariwala Okay, so a similar approach was being used earlier when the first ExperimentDTO was implemented by Ridham. Should I go ahead and update those return types and DB save types as well?

**Comment by VivekFitkariwala** [comment]
@ppratikcr7, not in this PR, create a separate ticket.

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**